File: app/src/main/java/com/tasomaniac/devwidget/data/WidgetAppDao.java
package com.tasomaniac.devwidget.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Transaction;

import java.util.List;

import io.reactivex.Flowable;

@Dao
public interface WidgetAppDao {

    @Transaction
    @Query("SELECT * FROM widget WHERE appWidgetId != -1")
    Flowable<List<WidgetAndPackageNames>> allWidgetsWithPackages();
}

File: app/src/main/java/com/tasomaniac/devwidget/data/WidgetAppDao.kt
package com.tasomaniac.devwidget.data

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Transaction

import io.reactivex.Flowable

@Dao
interface WidgetAppDao {

    @Transaction
    @Query("SELECT * FROM widget WHERE appWidgetId != -1")
    fun allWidgetsWithPackages(): Flowable<List<WidgetAndPackageNames>>
}
